### ISSUE TITLE
Execution on ide not possible

### ISSUE THREAD

**Issue by vincenzopalazzo** [user12043/JPomoTimer#1]
Hello,

I took a look at your project and I like the idea, I would like to open it with my ide and without the file of setting.json this is not possible and it is a bit difficult for me to rebuild it, it's not that you can solve this problem ?

Also I tried to download the zip file containing the executables but I'm not going to extract it correctly raises an error, I thought it was my problem but trying to extract the zip that generates github for the download of the project the problem does not persist.
Can you solve this problem too so I can try your project?

**Comment by user12043** [comment]
Hi,
The existence of "settings.json" file should not be a problem. I tried to download the code and open the project with my IDE (with Intellij IDEA) but there was no problem. Why it is not possible to open the project without the settings file? (Intellij idea is recognizing the maven project with pom.xml file.)

I downloaded zip file contains the executables and executed the program successfully. So could give more information about your error? I compressed and extracted the executables with 7-zip on windows and opened with "Ark" on linux also.

Thanks for your interest

**Comment by vincenzopalazzo** [comment]
Hello @user12043 ,
sorry I'm bad, I'll post my mistakes that I found.
![screenshot from 2018-11-07 23-10-53](https://user-images.githubusercontent.com/17150045/48164866-240c3200-e2e4-11e8-8a99-cd736fb1797c.png)

![screenshot from 2018-11-07 23-18-59](https://user-images.githubusercontent.com/17150045/48164872-27072280-e2e4-11e8-9bd8-d130f62c33c0.png)

Also it would not be easier to post executables directly without zipping them?

**Comment by user12043** [comment]
Hello @vincenzopalazzo 
Now I understand the problem about JSONException In below function from `ogr.user12043.jpomotimer.Utils`

    public static void loadSettings() {
            try {
                String fileContent = readSettingsFile();
                JSONObject jsonObject = new JSONObject(fileContent);
                // Some code here...
            } catch (IOException | JSONException e) {
                System.err.println("Failed to read settings file. Using default settings.");
                e.printStackTrace();
            }
    }

The `readSettingsFile()` function returns an empty string if the file does not exists. Then `fileContent` becomes `""`. Because of this, `new JSONObject(fileContent)` throws `JSONException` and this exception is catching by catch block. So this exception is not blocking the program. Just printing the error with the line `e.printStackTrace()`. So the problem is not actually a problem. But I think the program should not print the exception to stdout and I edited the code and committed now.



And about the error when you trying the unzip the binaries, the archive created with 7-zip on windows, I am using Linux as you and I am getting the same error `An error occurred when extracting files` on some `.zip` files. But I successfully extracted my executables without this error. I haven't any solution about the error. So I uploaded the `tar.gz` archive of the binaries of my project. This should help you.
